### ISSUE TITLE
DateRangePicker style breaks if rerenders are caused in a certain way

### ISSUE THREAD

**Issue by uthbees** [rsuite/rsuite#3800]
### What version of rsuite are you using?

5.60.2

### What version of React are you using?

18.3.1

### What version of TypeScript are you using (if any)?

N/A

### What browser are you using?

Firefox 125.0.2

### Describe the Bug

This is very strange. Using a certain combination of props, this code causes the second DateRangePicker calendar to shift below the first one:

```js
const [initialized, setInitialized] = useState(false);

  useEffect(() => {
    if (!initialized) {
      setInitialized(true);
    }
  }, [initialized]);
```

This is what it looks like:

![image](https://github.com/rsuite/rsuite/assets/33760679/7aac6019-f840-498d-8139-0aa389556977)

### Expected Behavior

The calendars should stay next to each other like they normally do.

### To Reproduce

This reproduces the bug:

```js
function App() {
  // This part is just to put the date picker near the right edge of the screen.
  return (
    <div style={{ display: "flex" }}>
      <div style={{ marginLeft: "auto" }}>
        <DatePickerComponent />
      </div>
    </div>
  );
}

function DatePickerComponent() {
  const calendarAnchor = useRef(null);

  const [initialized, setInitialized] = useState(false);

  useEffect(() => {
    if (!initialized) {
      setInitialized(true);
    }
  }, [initialized]);

  return (
    <div ref={calendarAnchor} style={{ position: "relative" }}>
      <DateRangePicker
        placement="bottomEnd"
        container={calendarAnchor.current || undefined}
        ranges={[
          {
            label: "qwertyuiop",
            value: [new Date(), new Date()],
            placement: "left",
          },
        ]}
      />
    </div>
  );
}
```

Here's a link to a codesandbox with this code: [https://codesandbox.io/p/sandbox/rsuite-daterangepicker-bug-tvjq98](https://codesandbox.io/p/sandbox/rsuite-daterangepicker-bug-tvjq98).

I'm really not sure why this is happening. Using a ref instead of state fixes the issue here, but there's apparently more than one place in our project that's triggering this, since it's still broken if I make the change in our full project. I think it's something that needs to be fixed on this side.

The problem apparently originated in v5.37.0, as it works fine in v5.36.0 but not v5.37.0. I looked through [the diff](https://github.com/rsuite/rsuite/compare/v5.36.0...v5.37.0), but I don't see anything that could be obviously causing this there. Really not sure what's going on here.

**Comment by simonguo** [comment]
Hi  @uthbees 

The reason for this problem should be that the specified container of DateRangePicker affects the width calculation when rendering CSS. There are two ways to solve it:

- Delete the container prop.
- Add a piece of CSS to solve it.

```css
.rs-picker-daterange-calendar-group {
  display: flex;
  flex-wrap: nowrap;
}
```
> https://codesandbox.io/p/sandbox/rsuite-daterangepicker-bug-forked-8n3d7m

**Comment by uthbees** [comment]
Nice, that indeed will fix it! Thank you!

**Comment by simonguo** [comment]
Fixed in [v5.61.0](https://github.com/rsuite/rsuite/releases/tag/v5.61.0).
